site_scons/components/__init__.py

```diff
@@ -221,50 +219,34 @@ def define_common(reqs):
     """common system component definitions"""
     reqs.define('cmocka', libs=['cmocka'], package='libcmocka-devel')
 
-    reqs.define('libunwind', libs=['unwind'], headers=['libunwind.h'],
-                package='libunwind-devel')
+    reqs.define('libunwind', libs=['unwind'], headers=['libunwind.h'], package='libunwind-devel')
 
     reqs.define('lz4', headers=['lz4.h'], package='lz4-devel')
 
-    reqs.define('valgrind_devel', headers=['valgrind/valgrind.h'],
-                package='valgrind-devel')
+    reqs.define('valgrind_devel', headers=['valgrind/valgrind.h'], package='valgrind-devel')
 
-    reqs.define('cunit', libs=['cunit'], headers=['CUnit/Basic.h'],
-                package='CUnit-devel')
+    reqs.define('cunit', libs=['cunit'], headers=['CUnit/Basic.h'], package='CUnit-devel')
 
-    reqs.define('python34_devel', headers=['python3.4m/Python.h'],
-                package='python34-devel')
-
-    reqs.define('libelf', headers=['libelf.h'], package='elfutils-libelf-devel')
-
-    reqs.define('tbbmalloc', libs=['tbbmalloc_proxy'], package='tbb-devel')
-
-    reqs.define('jemalloc', libs=['jemalloc'], package='jemalloc-devel')
-
-    reqs.define('boost', headers=['boost/preprocessor.hpp'],
-                package='boost-python36-devel')
+    reqs.define('boost', headers=['boost/preprocessor.hpp'], package='boost-python36-devel')
 
     reqs.define('yaml', headers=['yaml.h'], package='libyaml-devel')
 
     reqs.define('event', libs=['event'], package='libevent-devel')
 
-    reqs.define('crypto', libs=['crypto'], headers=['openssl/md5.h'],
-                package='openssl-devel')
+    reqs.define('crypto', libs=['crypto'], headers=['openssl/md5.h'], package='openssl-devel')
 
-    reqs.define('json-c', libs=['json-c'], headers=['json-c/json.h'],
-                package='json-c-devel')
+    reqs.define('json-c', libs=['json-c'], headers=['json-c/json.h'], package='json-c-devel')
 
-    if reqs.get_env('PLATFORM') == 'darwin':
-        reqs.define('uuid', headers=['uuid/uuid.h'])
-    else:
-        reqs.define('uuid', libs=['uuid'], headers=['uuid/uuid.h'],
-                    package='libuuid-devel')
+    reqs.define('uuid', libs=['uuid'], headers=['uuid/uuid.h'], package='libuuid-devel')
+
+    reqs.define('hwloc', libs=['hwloc'], headers=['hwloc.h'], package='hwloc-devel')
+
+    reqs.define('protobufc', libs=['protobuf-c'], headers=['protobuf-c/protobuf-c.h'],
+                package='protobuf-c-devel')
 
 
 def define_ompi(reqs):
     """OMPI and related components"""
-    reqs.define('hwloc', headers=['hwloc.h'], libs=['hwloc'],
-                package='hwloc-devel')
```

Comment:
I don't like removing these.   With package= line, it doesn't try to build hwloc or protobufc.  It is there primarily for configure checks.   Now, if you try to build daos without protobufc or hwloc installed, you will get a build failure.   I think it's better to do the configure checks up front for things that are not always automatically installed in the distro.

Reply:
I simply moved this so it was in defined in the same place as others, the check should be exactly the same.  If you try to build daos without protobufc or hwloc installed then it'll tell you what package to install (assuming you're on el8)

Reply:
> it'll tell you what package to install

On Frontera, we can't install any packages to system paths because we don't have root permission.
Frontera is technically not a production environment, but if it's not too much maintenance to allow scons to optionally build for us...